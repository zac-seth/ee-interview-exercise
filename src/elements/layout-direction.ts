export enum LayoutDirection {
  ROW = `row`,
  COLUMN = `column`,
}
