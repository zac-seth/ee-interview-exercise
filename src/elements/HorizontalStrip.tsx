import { FC } from 'react';

import { LayoutDirection } from './layout-direction';
import { Layout } from './Layout.style';

export type HorizontalStripProps = { length: number };

export const HorizontalStrip: FC<HorizontalStripProps> = ({
  children,
  length
}) => (
  <Layout
    layoutDirection={LayoutDirection.ROW}
    length={length}
    data-testid='horizontal-strip-layout'
  >
    {children}
  </Layout>
);
