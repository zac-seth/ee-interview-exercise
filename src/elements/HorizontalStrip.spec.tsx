import { render, screen } from '@testing-library/react';

import { HorizontalStrip } from './HorizontalStrip';
import { LayoutDirection } from './layout-direction';

const LENGTH = 3;

const makeChildren = (length: number): JSX.Element[] =>
  Array.from({ length }, (_v, i) => (
    <div data-testid={`item-${i}`} key={`key-${i}`}>
      value {i}
    </div>
  ));

const makeStrip = () => (
  <HorizontalStrip length={LENGTH}>{makeChildren(3)}</HorizontalStrip>
);

describe(`HorizontalStrip`, () => {
  it(`Should render the HorizontalStrip with the list of given child items`, () => {
    render(makeStrip());

    expect(screen.getByTestId(`horizontal-strip-layout`)).toMatchSnapshot();
  });

  it(`Should render the grid-template-column direction attribute with the expected value`, () => {
    render(makeStrip());

    const element = screen.getByTestId(`horizontal-strip-layout`);

    expect(element).toHaveStyleRule(`grid-template-columns`, `repeat(3,1fr)`);
  });

  it(`Should render the children in the HorizontalStrip`, async () => {
    render(makeStrip());

    const children = await screen.findAllByTestId(`item`, { exact: false });

    expect(children.length).toBe(3);
  });
});
