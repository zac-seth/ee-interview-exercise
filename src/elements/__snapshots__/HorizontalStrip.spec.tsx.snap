// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalStrip Should render the HorizontalStrip with the list of given child items 1`] = `
.c0 {
  display: grid;
  grid-gap: 0.25rem;
  margin: 0;
  padding: 0;
  grid-template-columns: repeat(3,1fr);
}

<div
  class="c0"
  data-testid="horizontal-strip-layout"
>
  <div
    data-testid="item-0"
  >
    value 
    0
  </div>
  <div
    data-testid="item-1"
  >
    value 
    1
  </div>
  <div
    data-testid="item-2"
  >
    value 
    2
  </div>
</div>
`;
